corehq/apps/data_dictionary/views.py

```diff
@@ -87,19 +88,20 @@ def data_dictionary_json(request, domain, case_type_name=None):
             "is_deprecated": case_type.is_deprecated,
             "module_count": module_count,
             "properties": [],
-            "contains_case_data": len(used_props) > 0,
+            "is_safe_to_delete": len(used_props) == 0,
         }
         grouped_properties = {
             group: [
                 {
+                    'id': prop.id,
                     'description': prop.description,
                     'label': prop.label,
                     'fhir_resource_prop_path': fhir_resource_prop_by_case_prop.get(
                         prop
                     ),
                     'name': prop.name,
                     'deprecated': prop.deprecated,
-                    'contains_case_data': prop.name in used_props,
+                    'is_safe_to_delete': prop.name not in used_props and prop.name != geo_case_prop,
```

Comment:
Thinking about someone in the future seeing this code here: Might they not wonder why the geo_case_property is excluded? I'm wondering if we should put this property in a list of properties that must not be deleted, but I'm not sold on this yet. Thoughts?

Reply:
I share this sentiment of not having a seemingly random additional check.

Reply:
That's a valid point. No arguments against putting it in a list, however I think the problem would still remain of future devs wondering why the properties in the list can't be deleted. @SmittieC What do you think about addressing this with a comment or in the documentation itself rather?

Reply:
With a comment yes, in docs, no. Reason being that dev's wouldn't read the docs necessarily.

Reply:
>  Reason being that dev's wouldn't read the docs necessarily.

That is true. The other benefit of a code comment would be that devs can get immediate clarity on what's going on instead of having to go digging through documentation.

Given the above, I'll add in a comment.